tests/test_check_parallel.py

```diff
@@ -174,8 +177,22 @@ def teardown_class(self):
 
     def test_worker_initialize(self) -> None:
         linter = PyLinter(reporter=Reporter())
-        worker_initialize(linter=linter)
-        assert pylint.lint.parallel._worker_linter == linter
+        worker_initialize(linter=dill.dumps(linter))
+        assert isinstance(pylint.lint.parallel._worker_linter, type(linter))
+
+    def test_worker_initialize_pickling(self) -> None:
+        """Test that we can pickle objects that standard pickling in multiprocessing can't.
+
+        See:
+        https://stackoverflow.com/questions/8804830/python-multiprocessing-picklingerror-cant-pickle-type-function
+        https://github.com/PyCQA/pylint/pull/5584
+        """
+        linter = PyLinter(reporter=Reporter())
+        linter.attribute = argparse.ArgumentParser()  # type: ignore[attr-defined]
+        pool = multiprocessing.Pool(  # pylint: disable=consider-using-with
+            2, initializer=worker_initialize, initargs=[dill.dumps(linter)]
+        )
+        pool.imap_unordered(print, [1, 2])
```

Comment:
Shouldn't we assert something in this test ?

Reply:
This test will fail without using `dill` as `argparse.ArgumentParser` won't be able to be pickled. This is a bit of a silly test as the test uses `dill.dumps` itself, but should serve as a guard against anybody trying to remove `dill` as a dependency in the future without thinking of the impact on "pickle-bility".